ChaosMod/Effects/db/Player/PlayerRandomVehSeat.cpp

```diff
@@ -23,7 +23,19 @@ static void OnStart()
 		{
 			Ped playerPed = PLAYER_PED_ID();
 
-			Vehicle veh = vehs[g_random.GetRandomInt(0, vehs.size() - 1)];
+			Vehicle veh;
+			for (int i = 0; i < 100; i++)
```

Comment:
Why 100? shouldn't this loop until a suitable vehicle is found?

Reply:
It's more or less a failsafe so it won't loop forever if something goes wrong. During testing I haven't seen it loop more than 5 times to find a suitable vehicle tbh.

Reply:
Yeah we don't want it to get stuck when "no traffic" is active ;)
Instead of 100 we could take the vehs.size() though.

Reply:
Good idea, see new commit!

Reply:
The problem now is that if there's a very small amount of valid vehicles there's a high chance that it won't find them. For example if there's one valid vehicle and one invalid vehicle there's a 25% chance that it checks the invalid vehicle both times and never finds the valid one. Something like 10 loops should be fine for performance and not missing anything. Another option is to remove invalid vehicles from the vector when they're detected, then the loop could be equal to the vector size with no chance of missing valid vehicles.

Also the current code loops one time fewer than the amount of vehicles so it's going to miss valid vehicles even more often.

Reply:
I would move that check to the top, checking it every vehicle before moving it to the array of possible vehicles. This way you are sure to have a valid vehicle before selecting a random one

Reply:
Thanks for the feedback everyone. Pushed a new commit moving the checks to the vehicle array filling code.